File: back/app/schemas/company.py
from marshmallow import Schema, fields, validate


class CompanyDirectorRefSchema(Schema):
    id = fields.Int(dump_only=True)
    fio = fields.Str(dump_only=True)
    login = fields.Str(dump_only=True)
    avatar_path = fields.Str(dump_only=True, allow_none=True)


class CompanySettingsSchema(Schema):
    uses_yougile = fields.Bool(load_default=False)
    uses_stages = fields.Bool(load_default=False)
    uses_calls = fields.Bool(load_default=True)


class CompanySchema(Schema):
    id = fields.Int(dump_only=True)
    name = fields.Str(dump_only=True)
    description = fields.Str(dump_only=True, allow_none=True)
    is_active = fields.Bool(dump_only=True)
    settings = fields.Dict(dump_only=True)
    director = fields.Nested(CompanyDirectorRefSchema, dump_only=True, allow_none=True)
    director_id = fields.Int(dump_only=True, allow_none=True)
    created_at = fields.DateTime(dump_only=True)
    # Виртуальные поля для таблицы — заполняются обработчиком списка.
    employees_count = fields.Int(dump_only=True)
    tasks_count = fields.Int(dump_only=True)


class CompanyCreateSchema(Schema):
    name = fields.Str(required=True, validate=validate.Length(min=1, max=255))
    description = fields.Str(load_default=None, allow_none=True)
    director_id = fields.Int(load_default=None, allow_none=True)
    is_active = fields.Bool(load_default=True)
    settings = fields.Nested(CompanySettingsSchema, load_default=dict)


class CompanyUpdateSchema(Schema):
    name = fields.Str(validate=validate.Length(min=1, max=255))
    description = fields.Str(allow_none=True)
    director_id = fields.Int(allow_none=True)
    is_active = fields.Bool()
    settings = fields.Nested(CompanySettingsSchema, partial=True)


class CompanyToggleActiveSchema(Schema):
    is_active = fields.Bool(required=True)

File: back-go/auth/internal/domain/company.go
package domain

import "time"

// DefaultCompanySettings — дефолтные настройки рабочего процесса компании
// (DEFAULT_SETTINGS из back/app/models/company.py). Свежая map на каждый
// вызов — настройки компаний не должны делить ссылки.
func DefaultCompanySettings() map[string]any {
	return map[string]any{
		"uses_yougile": false,
		"uses_stages":  false,
		"uses_calls":   true,
		// Режим «Мой Groove» (геймификация-питомцы) включён по умолчанию.
		"uses_groove": true,
		// Выходные дни компании: 0=Пн … 6=Вс (Python date.weekday()).
		"weekend_days": []any{5, 6},
	}
}

// DefaultWeekend — суббота и воскресенье (DEFAULT_WEEKEND из utils/workweek.py).
var DefaultWeekend = []int{5, 6}

// CompanyDirector — корневой Руководитель в объёме CompanyDirectorRefSchema.
type CompanyDirector struct {
	ID         int64
	FIO        string
	Login      string
	AvatarPath *string
}

// Company — компания целиком (домен управления компаниями; CompanyRef из
// user.go остаётся узким срезом для клеймов токена).
type Company struct {
	ID          int64
	Name        string
	Description *string
	DirectorID  *int64
	Director    *CompanyDirector
	IsActive    bool
	Settings    map[string]any
	CreatedAt   time.Time
	InviteCode  *string
}

// CompanyStats — счётчики для строки таблицы Компаний.
type CompanyStats struct {
	Employees int
	Tasks     int
}
